src/Runner.Worker/JobRunner.cs

```diff
@@ -106,6 +106,10 @@ public async Task<TaskResult> RunAsync(Pipelines.AgentJobRequestMessage message,
 
                 jobContext.SetRunnerContext("os", VarUtil.OS);
 
+                var runnerSettings = HostContext.GetService<IConfigurationStore>().GetSettings();
```

Comment:
we might need to re-fetch these info from the server since the runner/group can be renamed/moved via API or UI.

Reply:
Right, I was just looking back at our earlier talk about this. I thought we fetch this info from `.runner` and I was gonna check if that updates to reflect changes in runner_group and name.

I'm gonna check how and when it changes and see where it's best to re-fetch.

Reply:
We can just fetch them at the beginning of the job.

Reply:
or do it on the listener side and updates the `.runner` file

Reply:
When we `RenewJobRequestAsync` on the listener side, we do get the latest runner name back, we might need to update the server to return the runner group back as well, so we can avoid another round trip to server.

Reply:
Updating `.runner` seems like the right way. I did see the problem with runner_groups, but can you also change runner names?

Reply:
technically, you can via REST api.

Reply:
I could only find `GET` and `DELETE` for the runner.
Also, I removed adding the `runner_group` from this PR.